src/http/web_socket.cr

```diff
@@ -100,52 +101,109 @@ class HTTP::WebSocket
     @ws.close(message)
   end
 
-  def run
+  enum Opcode
+    Text
+    Binary
+    Close
+    Ping
+    Pong
+  end
+
+  struct Message
+    getter opcode
+
+    def initialize(@opcode : Opcode, @data : String | Bytes | Nil = nil)
```

Comment:
In what case can `@data` be nil?

Reply:
`:close` without a message?

Reply:
That should be empty String I think.

Reply:
True, I think the data should always be an empty String or empty Bytes...